app/components/search.tsx

```diff
@@ -1,48 +1,59 @@
 import {useState, useEffect, ChangeEventHandler} from 'react'
-import {tensor2d, matMul, Tensor2D} from '@tensorflow/tfjs'
-import {load, UniversalSentenceEncoder} from '@tensorflow-models/universal-sentence-encoder'
-import debounce from 'lodash/debounce'
 
 type Question = {
   title: string
-  normalized: string
+  normalized?: string
 }
 
 type SearchResult = Question & {
   score: number
 }
 
-type SearchProps = {
-  isReady: boolean
-  questions: Question[]
-  encodings: Tensor2D | []
-  langModelPromise: Promise<UniversalSentenceEncoder>
-}
+var tfWorker: Worker
 
 export default function Search() {
-  const [searchProps, setSearchProps] = useState<SearchProps>({
-    questions: [],
-    isReady: false,
-    encodings: [],
-    langModelPromise: load(), // load universal sentence encoder model
-  })
+  const [isReady, setReady] = useState<boolean>(false)
   const [baselineSearchResults, setBaselineSearchResults] = useState<SearchResult[]>([])
   const [searchResults, setSearchResults] = useState<SearchResult[]>([])
   const [showResults, setShowResults] = useState(false)
 
-  useEffect(() => {
-    initSearch().then((props) => {
-      console.debug(props)
-      setSearchProps((current) => ({...current, ...props}))
-    })
-  }, [])
+  const handleWorker = (event) => {
+    // read and print out the incoming data
+    const {data} = event
+
+    // 1st time should be init search
+    // TODO: doesn't seem to be setting isReady correctly
+    if (data.isReady) {
+      console.log('setReady to', data.isReady)
+      setReady(data.isReady)
+    }
+    // else results from run search
+    if (data.searchResults) {
+      console.log('setSearchResults to', data.searchResults)
+      setSearchResults(data.searchResults)
+    }
+  }
 
-  const handleChange = debounce((value: string) => {
-    console.debug('searching for:', value)
-    runBaselineSearch(value, searchProps).then(setBaselineSearchResults)
+  const initWorker = () => {
+    // create worker thread
+    if (window.Worker) {
+      if (typeof tfWorker == 'undefined') {
+        tfWorker = new Worker('/tfWorker.js')
+      }
 
-    runSemanticSearch(value, searchProps).then((r) => r && setSearchResults(r))
-  }, 300)
+      // any other messages is likely from runSearch
+      tfWorker.addEventListener('message', handleWorker)
+    } else {
+      console.log('Sorry! No Web Worker support.')
+    }
+  }
+
+  useEffect(() => initWorker(), [])
```

Comment:
using hooks inside the init => I moved `tsWorker` to a component instance `tsWorkerRef.current` for now, that will work the same while we have only 1 search, and should be fine in case we add multiple search components in the future (each having separate worker => each worker importing twjs separetely, but each only once, so no problems expected)